### PR TITLE
chore: lock the databend-sqlalchemy version

### PR DESCRIPTION
**Pull request by hantmac**
### SUMMARY

Lock the databend-sqlalchemy version

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -143,6 +143,7 @@ def get_git_sha() -> str:
             "google-cloud-bigquery>=3.10.0",
         ],
         "clickhouse": ["clickhouse-connect>=0.5.14, <1.0"],
+        "databend": ["databend-sqlalchemy>=0.3.2, <0.5"],
```
Comment (villebro) [inline setup.py:146]: Is there some reason we want to support 0.3? Could we just update this to have the lower bound on the most recent verified version?

Comment (hantmac) [reply setup.py:146]: The versions less than 0.3 have some compatibility issues. 
Of course, we can use this  lower bound. You mean we just `"databend": ["databend-sqlalchemy>=0.3.2"]`?

**Comment by john-bodley** [comment]
Thanks @hantmac for the PR. Would you mind including a complete PR description per the template?

**Comment by hantmac** [comment]
> Thanks @hantmac for the PR. Would you mind including a complete PR description per the template?

HI @john-bodley , what is the mean of `description per the template`? Or would u like to describe it more detail? Thank you very much!

**Comment by john-bodley** [comment]
@hantmac when you create a PR it should have been auto populated with [this](https://github.com/apache/superset/blob/master/.github/PULL_REQUEST_TEMPLATE.md) template. You could copy-and-paste the markdown from said template and complete the necessary fields. This helps to provide reviewers et al. with the necessary information.

**Comment by hantmac** [comment]
> @hantmac when you create a PR it should have been auto populated with [this](https://github.com/apache/superset/blob/master/.github/PULL_REQUEST_TEMPLATE.md) template. You could copy-and-paste the markdown from said template and complete the necessary fields. This helps to provide reviewers et al. with the necessary information.

Thanks! I will add the template with necessary information.

File: setup.py (hunk 1 above)
Comment (john-bodley) [reply setup.py:146]: @hantmac per [PyPI](https://pypi.org/project/databend-sqlalchemy/#history) the latest release of said package is `0.3.2` and thus the open upper limit of `0.5` seems somewhat arbitrary. Generally—assuming that the package is adhering to semver—placing an open upper limit of the major version (in this case `1.0`) is preferred as this _should_ prevent breaking changes.

Comment (hantmac) [reply setup.py:146]: > @hantmac per [PyPI](https://pypi.org/project/databend-sqlalchemy/#history) the latest release of said package is `0.3.2` and thus the open upper limit of `0.5` seems somewhat arbitrary. Generally—assuming that the package is adhering to semver—placing an open upper limit of the major version (in this case `1.0`) is preferred as this _should_ prevent breaking changes.

You mean I should set `databend-sqlalchemy>=0.3.2, <1.0` ?

**Review by john-bodley — APPROVED** [review]

**Review by aliblade — APPROVED** [review]

**Review by aliblade — APPROVED** [review]

**Pull request merged**